### ISSUE TITLE
Substitutors matches numeric character references while searching for hash index

### ISSUE THREAD

**Issue by simonmysun** [asciidoctor/asciidoctor#4393]
Hi, 

in 2.0.18,

```adoc
see <<Cub => Tiger>>

== Cub => Tiger
```

yields:

```html
<div class="paragraph">
<p>see <a href="Cub &.html#8658; Tiger">Cub &.html</a></p>
</div>
<div class="sect1">
<h2 id="_cub_tiger">Cub &#8658; Tiger</h2>
<div class="sectionbody">

</div>
</div>
```

The result should be:

```html
<div class="paragraph">
<p>see <a href="#_cub_tiger">Cub &#8658; Tiger</a></p>
</div>
<div class="sect1">
<h2 id="_cub_tiger">Cub &#8658; Tiger</h2>
<div class="sectionbody">

</div>
</div>
```

I have prepared a patch to fix it.

**Comment by mojavelinux** [comment]
Sounds reasonable. In fact, this comes up when a numeric character reference exists in the path of an interdocument xref too.

I'll go ahead and schedule this to be backported to 2.0.x.

**Comment by mojavelinux** [comment]
I just realized we solved another open issue with this fix, #3642. :tada:
